src/main/java/core/basesyntax/MyLinkedList.java

```diff
@@ -3,48 +3,158 @@
 import java.util.List;
 
 public class MyLinkedList<T> implements MyLinkedListInterface<T> {
+
+    private Node<T> firstNode;
+    private Node<T> lastNode;
+    private int count;
+
+    public MyLinkedList() {
+    }
+
     @Override
     public boolean add(T value) {
-        return false;
+        Node<T> temp = new Node<>(value, null, lastNode);
+        if (count == 0) {
+            firstNode = temp;
+        } else {
+            lastNode.next = temp;
+        }
+        lastNode = temp;
+        count++;
+        return true;
     }
 
     @Override
     public void add(T value, int index) {
-
+        if (index < 0 || index > count) {
+            throw new IndexOutOfBoundsException();
+        } else {
+            if (index == count) {
+                add(value);
+            } else {
+                Node<T> nextNode = getNodeByIndex(index);
+                Node<T> prewNode = nextNode.prev;
+                Node<T> newNode = new Node<>(value, nextNode, prewNode);
+                if (prewNode == null) {
+                    firstNode = newNode;
+                } else {
+                    prewNode.next = newNode;
+                }
+                nextNode.prev = newNode;
+                count++;
+            }
+        }
     }
 
     @Override
     public boolean addAll(List<T> list) {
-        return false;
+        for (T item : list) {
+            add(item);
+        }
+        return true;
     }
 
     @Override
     public T get(int index) {
-        return null;
+        if (indexIsExist(index)) {
+            Node<T> temp = getNodeByIndex(index);
+            return temp.item;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public T set(T value, int index) {
-        return null;
+        if (indexIsExist(index)) {
+            Node<T> temp = getNodeByIndex(index);
+            T oldValue = temp.item;
+            temp.item = value;
+            return oldValue;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public T remove(int index) {
-        return null;
+        if (indexIsExist(index)) {
+            Node<T> temp = getNodeByIndex(index);
+            unlink(temp);
+            return temp.item;
+        }
+        throw new IndexOutOfBoundsException();
     }
 
     @Override
     public boolean remove(T t) {
+        for (int i = 0; i < count; i++) {
+            if (getNodeByIndex(i).item == t || getNodeByIndex(i) != null
```

Comment:
should we call the method `getNodeByIndex(i)` three times? Let's do it only once